modules/core/src/v2/wallet.ts

```diff
@@ -517,90 +519,87 @@ export class Wallet {
+   * @param {Number} params.limit                for routeName === 'consolidate'
+   *                 params.maxNumUnspentsToUse  for routeName === 'fanout'
+   *                  - maximum number of unspents you want to use in the transaction
+   * Output parameters:
+   * @param {Number} params.numUnspentsToMake - the number of new unspents to make
+   *
    * @param callback
    * @returns txHex {String} the txHex of the incomplete transaction that needs to be signed by the user in the SDK
    */
-  consolidateUnspents(params: any = {}, callback?: NodeCallback<any>): Bluebird<any> {
-    const self = this;
+  manageUnspents(routeName: ManageUnspents, params: any = {}, callback?: NodeCallback<any>): Bluebird<any> {
     return co(function *() {
       common.validateParams(params, [], ['walletPassphrase', 'xprv'], callback);
 
       const reqId = new RequestTracer();
-      const keychain = yield self.baseCoin.keychains().get({ id: self._wallet.keys[0], reqId });
-      const filteredParams = _.pick(params, ['minValue', 'maxValue', 'minHeight', 'numUnspentsToMake', 'feeTxConfirmTarget', 'limit', 'minConfirms', 'enforceMinConfirmsForChange', 'feeRate', 'maxFeeRate', 'maxFeePercentage']);
-      self.bitgo.setRequestTracer(reqId);
-      const response = yield self.bitgo.post(self.url('/consolidateUnspents'))
+      const filteredParams = _.pick(params, [
+        'feeRate',
+        'maxFeeRate',
+        'maxFeePercentage',
+        'feeTxConfirmTarget',
+
+        'minValue',
+        'maxValue',
+        'minHeight',
+        'minConfirms',
+        'enforceMinConfirmsForChange',
+
+        routeName === 'consolidate' ? 'limit' : 'maxNumUnspentsToUse',
+        'numUnspentsToMake',
+      ]);
+      this.bitgo.setRequestTracer(reqId);
+      const response = yield this.bitgo.post(this.url(`/${routeName}Unspents`))
         .send(filteredParams)
         .result();
 
-      const transactionParams = _.extend({}, params, { txPrebuild: response, keychain: keychain });
-      const signedTransaction = yield self.signTransaction(transactionParams);
+      const keychain = yield this.baseCoin.keychains().get({ id: this._wallet.keys[0], reqId });
+      const transactionParams = _.extend({}, params, { txPrebuild: response, keychain });
+      const signedTransaction = yield this.signTransaction(transactionParams);
       const selectParams = _.pick(params, ['comment', 'otp']);
       const finalTxParams = _.extend({}, signedTransaction, selectParams);
 
-      self.bitgo.setRequestTracer(reqId);
-      return self.bitgo.post(self.baseCoin.url('/wallet/' + self._wallet.id + '/tx/send'))
+      this.bitgo.setRequestTracer(reqId);
+      return this.bitgo.post(this.baseCoin.url('/wallet/' + this._wallet.id + '/tx/send'))
         .send(finalTxParams)
         .result();
-    }).call(this).asCallback(callback);
+    }.bind(this)).call(this).asCallback(callback);
```

Comment:
good point

but why even `call(this)` then? `call()` is also an option (will use `global` as `this`)

Reply:
`call()` would also work, but would break if someone accidentally forgets to use `self`, as you mentioned before.

Longer term, we really just need to move to async/await and drop all the this-rebinding nonsense.

Reply:
I played around with some AST transforms for that problems, hit me up when you prioritize it